src/views/blog-post/blog-post.astro

```diff
@@ -46,20 +48,32 @@ const {
 	collection,
 	collectionPosts,
 	license,
+	usedComponents,
 } = Astro.props;
 
 const GHLink = `https://github.com/${siteMetadata.repoPath}/tree/main/content/${post.path}/index.md`;
+
+const scriptServerPaths = Object.keys(usedComponents).reduce((prev, key) => {
+	if (!usedComponents[key]) return prev;
+	prev.push(getComponentScriptPath(key).serverRelativePath);
+	return prev;
+}, [] as string[]);
```

Comment:
nit: consider passing a type argument to avoid `as` casting

```diff
- const scriptServerPaths = Object.keys(usedComponents).reduce((prev, key) => {
+ const scriptServerPaths = Object.keys(usedComponents).reduce<string[]>((prev, key) => {
	  if (!usedComponents[key]) return prev;
	  prev.push(getComponentScriptPath(key).serverRelativePath);
	  return prev;
+ }, []);
- }, [] as string[]);
```

Reply:
I don't prefer this syntax as it makes things harder to read IMO and breaks implicit return typing of the function itself.